### PR TITLE
Added test project and moved to Microsoft.CodeAnalysis v1.0.0

### LINKED ISSUE #11: Add tests.
We should incorporate tests into the project.

### PR DESCRIPTION
**Pull request by JoshVarty**
This will fix #11 and from now on we should be adding tests when fixing generator bugs such as #17 and #64.

I've used Kirill Osenkov's `MSBuildWorkspaceTestBase` class. You can see how they're using it internally here: http://source.roslyn.codeplex.com/#Roslyn.Services.UnitTests/WorkspaceTests/MSBuildWorkspaceTests.cs,148

It seems like a nice approach to quickly and easily create solutions without having to write a ton of boilerplate code.

This new version of Microsoft.CodeAnalysis depends on MEF 2 which is the `Microsoft.Composition` package on NuGet.

I still have to write tests for parameters and locals and will do so after #66 is merged.

### CODE REVIEW

File: src/SourceBrowser.Tests/CSharpTests.cs [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                            public void M1 () { }
+                        }")));
+
+            WorkspaceModel ws = new WorkspaceModel("Workspace1", "");
+            FolderModel fm = new FolderModel(ws, "Project1");
+
+            var document = solution.Projects.SelectMany(n => n.Documents).Where(n => n.Name == "Document1.cs").Single();
+            var linkProvider = new ReferencesourceLinkProvider();
+
+            var walker = new SourceBrowser.Generator.DocumentWalkers.CSWalker(fm, document, linkProvider);
```
Comment (AmadeusW) [inline src/SourceBrowser.Tests/CSharpTests.cs]: var walker = SourceBrowser.Generator.DocumentWalkers.WalkerSelector.GetWalker(fm, document, linkProvider);

Also, make SourceBrowser.Generator.DocumentWalkers.WalkerSelector a public class
Apply this change to other tests

**Comment by AmadeusW** [comment]
MSBuildWorkspaceTestClass looks really handy. Good job! :smile: 

I tried integrating this with Marcin's PR #70 but the checked out pull request branches are read only (which means that only he can adapt his code to this merge). Is it ok if we merge #70 first? Then we need to do the following:

1. make WalkerSelector a public class
2. use WalkerSelector throughout the test cases
3. use the tests to verify that the CSWalker works

**Comment by MarcinJuraszek** [comment]
You can add `InternalsVisibleTo` to the generator and make internal members visible from UnitTests project instead of making `WalkerSelector` public.

**Comment by JoshVarty** [comment]
Yeah, we'll just use InternalsVisibleTo and make WalkerSelector internal instead of private.

**Pull request merged**